packages/smart-contracts/test/contracts/BatchEthereumPayments.test.ts

```diff
@@ -0,0 +1,210 @@
+    expect(await provider.getBalance(batchAddress)).to.be.equal(0);
+  });
+
+  it('Should pay 2 payments with the exact amount', async function () {
+    beforeEthBalance = await provider.getBalance(payeeOne);
+    beforeEthBalance1 = await provider.getBalance(payeeTwo);
+    let totalAmout = BigNumber.from('53');
+
+    tx = await batch
+      .connect(owner)
+      .batchEthereumPaymentsWithReferenceAndFee(
+        [payeeOne, payeeTwo],
+        [20, 30],
+        [referenceExample1, referenceExample2],
+        [1, 2],
+        feeAddress,
+        {
+          value: totalAmout,
+        },
+      );
+    await tx.wait();
+
+    afterEthBalance = await provider.getBalance(payeeOne);
+    expect(afterEthBalance).to.be.equal(beforeEthBalance.add(20));
+
+    afterEthBalance1 = await provider.getBalance(payeeTwo);
+    expect(afterEthBalance1).to.be.equal(beforeEthBalance1.add(30));
+  });
+
+  it('Gas evaluation: Should pay multiple Ethereum payments', async function () {
+    let listTxs = [1, 12]; // [1, 4, 8, 12, 30, 100]; //
+    for (let i = 0; i < listTxs.length; i += 1) {
+      beforeEthBalance = await provider.getBalance(payeeTwo);
+      nbTxs = listTxs[i];
+      let [_, recipients, amounts, paymentReferences, feeAmounts] = getInputsBatchPayments(
+        nbTxs,
+        '_noTokenAddress',
+        payeeTwo,
+        amount,
+        referenceExample1,
+        feeAmount,
+      );
+      let totalAmount = BigNumber.from(((amount + feeAmount) * nbTxs).toString());
+
+      tx = await batch
+        .connect(owner)
+        .batchEthereumPaymentsWithReferenceAndFee(
+          recipients,
+          amounts,
+          paymentReferences,
+          feeAmounts,
+          feeAddress,
+          {
+            value: totalAmount,
+          },
+        );
+
+      const receipt = await tx.wait();
+      if (logInfos) {
+        console.log(`nbTxs= ${nbTxs}, gas consumption: `, receipt.gasUsed.toString());
+      }
+
+      afterEthBalance = await provider.getBalance(payeeTwo);
+      expect(afterEthBalance).to.be.equal(beforeEthBalance.add(amount * nbTxs));
+    }
+  });
+
+  it('Should revert and any payment is achieved because funds are not sufficient', async function () {
+    beforeEthBalance = await provider.getBalance(payeeOne);
+    beforeEthBalance1 = await provider.getBalance(payeeTwo);
+    let totalAmout = BigNumber.from('40');
+
+    await expect(
+      batch
+        .connect(owner)
+        .batchEthereumPaymentsWithReferenceAndFee(
+          [payeeOne, payeeTwo],
+          [20, 30],
+          [referenceExample1, referenceExample2],
+          [1, 2],
+          feeAddress,
+          {
+            value: totalAmout,
+          },
+        ),
+    ).to.be.reverted;
+
+    afterEthBalance = await provider.getBalance(payeeOne);
+    expect(afterEthBalance).to.be.equal(beforeEthBalance);
+
+    afterEthBalance1 = await provider.getBalance(payeeTwo);
+    expect(afterEthBalance1).to.be.equal(beforeEthBalance1);
```

Comment:
I would also test that the wallet's ETH balance is 0, and it might apply to other tests BTW